src/includes/traits/Plugin/AdminBarUtils.php

```diff
@@ -231,7 +231,7 @@ public function adminBarMenu(\WP_Admin_Bar &$wp_admin_bar)
                     'id'     => GLOBAL_NS.'-stats',
 
                     'title' => __('Cache Stats', SLUG_TD),
-                    'href'  => '#',
+                    'href'  => ''.esc_attr(add_query_arg(urlencode_deep(['page' => GLOBAL_NS.'-stats']), network_admin_url('/admin.php'))).'',
```

Comment:
@renzms You can make this a little cleaner by removing the double quotes; i.e., `=> ''.esc_attr(` on the beginning and end of this string would only be needed if there was something inside them; e.g., `=> 'something'.esc_attr(`. In this case there is not, so it can just be `=> esc_attr( ...`.